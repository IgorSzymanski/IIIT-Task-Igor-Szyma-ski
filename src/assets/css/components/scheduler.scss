@import '../variables';

$day-border-width: 1px;
$day-padding: $box-padding;
$day-radius: 30em;

$hours: 24;
$minutes: 60;
$columns: [days] auto;
    @for $h from 0 through ($hours - 1) {
    @for $m from 0 through ($minutes/10 - 1) {
        $columns: $columns + unquote(' [h#{$h}-#{$m*10}] 1fr');
    }
}

$columns: $columns + unquote('[h24-0] 6em [h25-0 actions]');