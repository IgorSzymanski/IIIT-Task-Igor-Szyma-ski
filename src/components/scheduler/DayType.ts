export type DayType = {
	date: Date | string
	day: number
	month: [number, string]
	dayOfTheWeek: [number, string, string]
}
