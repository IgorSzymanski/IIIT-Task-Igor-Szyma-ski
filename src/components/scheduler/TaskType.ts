export type TaskType = {
	id: number
	name: string
	logs: TaskLogType[]
}

export type TaskLogType = {
	id: number
	start: Date | string
	end: Date | string
	status: TaskLogStatusType
}

export type ReversedTaskType = {
	id: number
	name: string
}

export type ReversedLogType = TaskLogType & {
	id: number
	start: Date
	end: Date
	status: TaskLogStatusType
	task: ReversedTaskType
}

export type ReversedLogTypeDTO = {
	id?: number
	start: Date
	end: Date
	status: TaskLogStatusType
	task: string
}

export type TaskLogStatusType = 'accepted' | 'pending' | 'warning' | 'active'
