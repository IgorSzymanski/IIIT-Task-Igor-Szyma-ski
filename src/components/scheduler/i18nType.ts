export type i18nType = {
	localeCode: string
	months: [
		string,
		string,
		string,
		string,
		string,
		string,
		string,
		string,
		string,
		string,
		string,
		string,
	]
	days: [string, string, string, string, string, string, string]
	shortDays: [string, string, string, string, string, string, string]
}
