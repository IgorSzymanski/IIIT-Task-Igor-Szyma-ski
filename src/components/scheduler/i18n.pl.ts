import { i18nType } from './i18nType'

export default {
	months: [
		'styczeń',
		'luty',
		'marzec',
		'kwiecień',
		'maj',
		'czerwiec',
		'lipiec',
		'sierpień',
		'wrzesień',
		'październik',
		'listopad',
		'grudzień',
	],
	days: [
		'poniedziałek',
		'wtorek',
		'środa',
		'czwartek',
		'piątek',
		'sobota',
		'niedziela',
	],
	shortDays: ['pn', 'wt', 'śr', 'cz', 'pt', 'sb', 'nd'],
} as i18nType
