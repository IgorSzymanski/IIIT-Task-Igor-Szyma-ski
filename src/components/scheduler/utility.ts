import dayjs from 'dayjs'
import { i18nType } from './i18nType'
import { TaskType, ReversedLogType } from './TaskType'
import i18nPl from './i18n.pl'
import { DayType } from './DayType'

export type FlexDate = string | Date | Readonly<Date>

/**
 * Used to render rows of days, therefore returns non-negative numbers.
 * Counts days between two days.
 * If endDay is before startDay, returns 0.
 */
export const countNumberOfDays = (startDay: Date | Readonly<Date>) => (
	endDay: Date | Readonly<Date>,
) => {
	const diff =
		dayjs(endDay.toISOString()).diff(dayjs(startDay.toISOString()), 'day') + 1
	return diff > 0 ? diff : 0
}

/**
 * Adds days to Date.
 */
export const addDay = (startDay: Date | Readonly<Date>) => (days: number = 1) =>
	dayjs(startDay.toISOString()).add(days, 'day')

export const generateDays = (i18n: i18nType) => (
	startDay: Date | Readonly<Date>,
) => (numberOfDays: number): DayType[] =>
	Array.from(Array(numberOfDays).keys())
		.map(addDay(startDay))
		.map(d => ({
			date: d.toDate(),
			day: Number(d.format('D')),
			month: [Number(d.format('M')), i18nPl.months[Number(d.format('M'))]],
			year: Number(d.format('YYYY')),
			dayOfTheWeek: [d.day(), i18nPl.days[d.day()], i18nPl.shortDays[d.day()]],
		}))

/**
 * Finds the earliest Date for 'isBefore' parameter, or latest for 'isAfter' parameter.
 */
export const findDate = (comp: 'isBefore' | 'isAfter') => (dates: Date[]) =>
	dates.length
		? dates
				.map(date => dayjs(date))
				.sort((a, b) => (a[comp](b) ? -1 : 1))[0]
				.toDate()
		: undefined

export const findEarliestDate = findDate('isBefore')
export const findLatestDate = findDate('isAfter')

/**
 * Converts TaskType to ReversedLogType
 */
export const reverseTaskData = (tasks: TaskType[]): ReversedLogType[] =>
	tasks.flatMap(task =>
		task.logs.map(log => ({
			...log,
			task: { id: task.id, name: task.name },
			end: toDate(log.end),
			start: toDate(log.start),
		})),
	)

/**
 * Returns Array of Date of start and end times of tasks.
 */
export const getAllDatesFromTasks = (tasks: TaskType[]): Date[] =>
	tasks
		.flatMap(task => task.logs.flatMap(log => [log.start, log.end]))
		.filter(date => !!date)
		.map(date => new Date(date))

/**
 * Converts String, Date, and readonly Date to Date.
 */
export const toDate = (date: FlexDate) =>
	typeof date === 'string'
		? new Date(date)
		: date instanceof Date
		? date
		: new Date('')

/**
 * Cheks if two dates are the same say. Hour may vary.
 */
export const isTheSameDay = (dateA: FlexDate) => (dateB: FlexDate) => {
	const A = dayjs(toDate(dateA))
	const B = dayjs(toDate(dateB))
	return A.isSame(B, 'year') && A.isSame(B, 'month') && A.isSame(B, 'day')
}

/**
 * Reverses Tasks and filters them by day.
 */
export const getReversedLogsForDate = (tasks: TaskType[]) => (date: FlexDate) =>
	reverseTaskData(tasks).filter(log => isTheSameDay(log.start)(date))

/**
 * Gets hour of date. Ex. for '2019-07-08T06:48:00Z' it's 6.
 */
export const getHour = (date: FlexDate) =>
	Number(dayjs(toDate(date)).format('H'))

/**
 * Gets minute of date. Ex. for '2019-07-08T06:48:00Z' it's 48.
 */
export const getMinute = (date: FlexDate) =>
	Number(dayjs(toDate(date)).format('m'))

/**
 * Gets minute of date. Ex. for '2019-07-08T06:48:00Z' it's 50.
 */
export const getRoundedMinute = (date: FlexDate) =>
	Math.floor(Number(dayjs(toDate(date)).format('m')) / 10) * 10

/**
 * Counts difference between two dates in minutes.
 */
export const countTimeDifference = (dateA: Date) => (dateB: Date) =>
	dayjs(dateA).diff(dayjs(dateB), 'minute')

/**
 * Sums time of logs and returns in minutes.
 */
export const countLogTime = (logs: ReversedLogType[]) =>
	logs
		.filter(log => log.status !== 'warning')
		.map(log => countTimeDifference(log.end)(log.start))
		.reduce((a, b) => a + b, 0)

/**
 * Formats minutes in HH:mm string format.
 */
export const formatMinutes = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes - hours * 60
	return `${pad(2)(hours)}:${pad(2)(minutes)}`
}

/**
 * Adds leading zeros to a number.
 */
export const pad = (size: number) => (number: number) => {
	let s = String(number)
	while (s.length < (size || 2)) {
		s = '0' + s
	}
	return s
}

export const parseDate = (date: string) =>
	dayjs(date, { format: 'DD.MM.YYYY' }).toDate()

export const combineDateAndHour = (date: Date | null) => (
	hour: Date | null,
) => {
	if (!date || !hour) {
		return
	}
	const dHour = dayjs(hour)
	return dayjs(date)
		.set('hour', dHour.hour())
		.set('minute', dHour.minute())
		.set('millisecond', dHour.millisecond())
		.toDate()
}

export const isDateAfter = (dateA: Date | null) => (dateB: Date | null) =>
	dateA && dateB ? dayjs(dateA).isAfter(dayjs(dateB)) : false

export const createDate = (day: number) => (month: number) => (year: number) =>
	dayjs(`${pad(4)(year)}-${pad(2)(month)}-${pad(2)(day)}`).toDate()

export const doDateRangesOverlap = (datesA: [FlexDate, FlexDate]) => (
	datesB: [FlexDate, FlexDate],
) => {
	const A = datesA.map(toDate).map(d => dayjs(d))
	const B = datesB.map(toDate).map(d => dayjs(d))
	return A.some(d => B[0].isBefore(d) && B[1].isAfter(d))
}
