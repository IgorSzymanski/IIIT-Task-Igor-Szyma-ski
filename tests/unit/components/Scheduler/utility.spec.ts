import { countNumberOfDays } from '@/components/scheduler/utility'
import { addDay } from '@/components/scheduler/utility'
import { generateDays } from '@/components/scheduler/utility'
import i18nPl from '@/components/scheduler/i18n.pl'
import {
	findDate,
	findEarliestDate,
	findLatestDate,
	reverseTaskData,
	getAllDatesFromTasks,
	toDate,
	isTheSameDay,
	getReversedLogsForDate,
} from '@/components/scheduler/utility'
import { TaskType, ReversedLogType } from '@/components/scheduler/TaskType'

describe('countNumberOfDays()', () => {
	it('Properly counts number of days needed to render scheduler', () => {
		expect(
			countNumberOfDays(new Date('1991-01-06'))(new Date('1991-01-16')),
		).toEqual(11)
	})

	it('Properly counts number of days when start and end days are the same', () => {
		expect(countNumberOfDays(new Date())(new Date())).toEqual(1)
	})

	it('Returns 0 when start day is after end day.', () => {
		expect(
			countNumberOfDays(new Date('1991-01-16'))(new Date('1991-01-6')),
		).toEqual(0)
		expect(
			countNumberOfDays(new Date('1991-01-7'))(new Date('1991-01-6')),
		).toEqual(0)
	})
})

describe('addDay()', () => {
	it('Properly adds positive days to a date', () => {
		expect(addDay(new Date('1991-01-6'))(1).toISOString()).toEqual(
			new Date('1991-01-7').toISOString(),
		)
		expect(addDay(new Date('1991-01-6'))(3).toISOString()).toEqual(
			new Date('1991-01-9').toISOString(),
		)
		expect(addDay(new Date('1991-01-6'))(365).toISOString()).toEqual(
			new Date('1992-01-6').toISOString(),
		)
	})

	it('Properly adds 0 days to a date', () => {
		expect(addDay(new Date('1991-01-6'))(0).toISOString()).toEqual(
			new Date('1991-01-6').toISOString(),
		)
	})

	it('Properly substracts days from a date', () => {
		expect(addDay(new Date('1991-01-6'))(-3).toISOString()).toEqual(
			new Date('1991-01-3').toISOString(),
		)

		expect(addDay(new Date('1991-01-6'))(-5).toISOString()).toEqual(
			new Date('1991-01-1').toISOString(),
		)

		expect(addDay(new Date('1991-01-6'))(-365).toISOString()).toEqual(
			new Date('1990-01-6').toISOString(),
		)
	})
})

describe('generateDays()', () => {
	const startDate = new Date('1991-01-1')
	const numberOfDays = 30
	const days = generateDays(i18nPl)(startDate)(numberOfDays)
	it('Generates correct number of days', () => {
		expect(days.length).toEqual(30)
	})

	it('Each element matches snapshot', () => {
		days.forEach(day => {
			expect(day).toMatchSnapshot({
				date: expect.any(Date),
				day: expect.any(Number),
				month: expect.any(Array),
				year: expect.any(Number),
				dayOfTheWeek: expect.any(Array),
			})

			expect(day.month).toMatchSnapshot([
				expect.any(Number),
				expect.any(String),
			])

			expect(day.dayOfTheWeek).toMatchSnapshot([
				expect.any(Number),
				expect.any(String),
				expect.any(String),
			])
		})
	})

	it('First day matches the start day', () => {
		expect(new Date(days[0].date).toISOString()).toEqual(
			startDate.toISOString(),
		)
	})
	it('Last day is [start day] + [number of days]', () => {
		expect(new Date(days[numberOfDays - 1].date).toISOString()).toEqual(
			addDay(startDate)(numberOfDays - 1).toISOString(),
		)
	})
})

describe('findDate()', () => {
	const earliest = new Date('1990-01-7')
	const mid = new Date('1991-01-6')
	const latest = new Date('1992-01-4')
	const dates = [mid, latest, earliest]
	const empty: Date[] = []

	it('Finds the earliest date in array', () => {
		expect(findDate('isBefore')(dates)).toEqual(earliest)
		expect(findEarliestDate(dates)).toEqual(earliest)
	})

	it('Finds the latest date in array', () => {
		expect(findDate('isAfter')(dates)).toEqual(latest)
		expect(findLatestDate(dates)).toEqual(latest)
	})

	it('Returns undefined if array is empty', () => {
		expect(findEarliestDate(empty)).toBeUndefined()
		expect(findLatestDate(empty)).toBeUndefined()
	})
})

const tasks: TaskType[] = [
	{
		id: 1,
		name: 'Task 1',
		logs: [
			{
				id: 1,
				start: new Date('2019-07-02T12:28:10Z'),
				end: new Date('2019-07-02T14:28:12Z'),
				status: 'accepted',
			},
			{
				id: 2,
				start: new Date('2019-07-04T16:59:00Z'),
				end: new Date('2019-07-04T18:03:08Z'),
				status: 'pending',
			},
		],
	},
	{
		id: 2,
		name: 'Task 2',
		logs: [
			{
				id: 3,
				start: new Date('2019-07-08T06:48:00Z'),
				end: new Date('2019-07-08T09:37:22Z'),
				status: 'accepted',
			},
		],
	},
]

const reversedLogs: ReversedLogType[] = [
	{
		id: 1,
		start: new Date('2019-07-02T12:28:10Z'),
		end: new Date('2019-07-02T14:28:12Z'),
		status: 'accepted',
		task: {
			id: 1,
			name: 'Task 1',
		},
	},
	{
		id: 2,
		start: new Date('2019-07-04T16:59:00Z'),
		end: new Date('2019-07-04T18:03:08Z'),
		status: 'pending',
		task: {
			id: 1,
			name: 'Task 1',
		},
	},
	{
		id: 3,
		start: new Date('2019-07-08T06:48:00Z'),
		end: new Date('2019-07-08T09:37:22Z'),
		status: 'accepted',
		task: {
			id: 2,
			name: 'Task 2',
		},
	},
]

describe('reverseTaskData()', () => {
	it('Properly reverse data', () => {
		expect(reverseTaskData(tasks)).toEqual(reversedLogs)
	})
})

describe('getAllDatesFromTasks()', () => {
	const allDates = getAllDatesFromTasks(tasks)
	it('Has proper number of dates', () => {
		expect(allDates.length).toEqual(6)
	})

	it('First date matches', () => {
		expect(allDates[0]).toEqual(new Date('2019-07-02T12:28:10Z'))
	})

	it('Last date matches', () => {
		expect(allDates[5]).toEqual(new Date('2019-07-08T09:37:22Z'))
	})
})

describe('isTheSameDay()', () => {
	it('When date is of Date type', () => {
		expect(
			isTheSameDay(new Date('2019-07-08T00:00:00Z'))(
				new Date('2019-07-08T06:48:00Z'),
			),
		).toBe(true)

		expect(
			isTheSameDay(new Date('2019-07-08T00:00:00Z'))('2019-07-08T06:48:00Z'),
		).toBe(true)

		expect(
			isTheSameDay(new Date('2018-07-08T00:00:00Z'))(
				new Date('2019-07-08T06:48:00Z'),
			),
		).toBe(false)

		expect(
			isTheSameDay(new Date('2019-08-08T00:00:00Z'))('2019-07-08T06:48:00Z'),
		).toBe(false)
	})

	it('When date is of String type', () => {
		expect(
			isTheSameDay('2019-07-08T00:00:00Z')(new Date('2019-07-08T06:48:00Z')),
		).toBe(true)

		expect(isTheSameDay('2019-07-08T00:00:00Z')('2019-07-08T06:48:00Z')).toBe(
			true,
		)

		expect(
			isTheSameDay('2018-07-08T00:00:00Z')(new Date('2019-07-08T06:48:00Z')),
		).toBe(false)

		expect(isTheSameDay('2019-08-08T00:00:00Z')('2019-07-08T06:48:00Z')).toBe(
			false,
		)
	})

	it('When date of String type is invalid', () => {
		expect(isTheSameDay('abc')(new Date('2019-07-08T06:48:00Z'))).toBe(false)

		expect(isTheSameDay('abc')('abc')).toBe(false)
	})
})

describe('toDate()', () => {
	it('When argument is valid date String or Date, return corresponding date', () => {
		expect(toDate('2019-07-08T06:48:00Z').toISOString()).toBe(
			new Date('2019-07-08T06:48:00Z').toISOString(),
		)

		expect(toDate(new Date('2019-07-08T06:48:00Z')).toISOString()).toBe(
			new Date('2019-07-08T06:48:00Z').toISOString(),
		)
	})

	it('When date String is invalid, returns invalid instance of Date', () => {
		const abc = toDate('abc')
		expect(abc instanceof Date).toBe(true)
		expect(isFinite(abc.getTime())).toBe(false)
	})
})

describe('getReversedLogsForDate()', () => {
	it('When logs for the day exist, return Array of ReversedLogs', () => {
		expect(
			getReversedLogsForDate(tasks)('2019-07-02T12:28:10Z'),
		).toStrictEqual([reversedLogs[0]])

		expect(
			getReversedLogsForDate(tasks)(new Date('2019-07-04T16:59:00Z')),
		).toStrictEqual([reversedLogs[1]])

		expect(
			getReversedLogsForDate(tasks)('2019-07-08T06:48:00Z'),
		).toStrictEqual([reversedLogs[2]])
	})

	it('When there are no logs for the day, return an empty Array', () => {
		expect(getReversedLogsForDate(tasks)('1991-07-08T06:48:00Z')).toStrictEqual(
			[],
		)
	})
})
