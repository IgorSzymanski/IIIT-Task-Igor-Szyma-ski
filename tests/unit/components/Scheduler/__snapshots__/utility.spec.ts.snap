// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateDays() Each element matches snapshot 1`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 2`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 3`] = `
Object {
  "0": 2,
  "1": "Sroda",
  "2": "Sr",
}
`;

exports[`generateDays() Each element matches snapshot 4`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 5`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 6`] = `
Object {
  "0": 3,
  "1": "Czwartek",
  "2": "Cz",
}
`;

exports[`generateDays() Each element matches snapshot 7`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 8`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 9`] = `
Object {
  "0": 4,
  "1": "Piątek",
  "2": "Pt",
}
`;

exports[`generateDays() Each element matches snapshot 10`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 11`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 12`] = `
Object {
  "0": 5,
  "1": "Sobota",
  "2": "Sb",
}
`;

exports[`generateDays() Each element matches snapshot 13`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 14`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 15`] = `
Object {
  "0": 6,
  "1": "Niedziela",
  "2": "Nd",
}
`;

exports[`generateDays() Each element matches snapshot 16`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 17`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 18`] = `
Object {
  "0": 0,
  "1": "Poniedziałek",
  "2": "Pn",
}
`;

exports[`generateDays() Each element matches snapshot 19`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 20`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 21`] = `
Object {
  "0": 1,
  "1": "Wtorek",
  "2": "Wt",
}
`;

exports[`generateDays() Each element matches snapshot 22`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 23`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 24`] = `
Object {
  "0": 2,
  "1": "Sroda",
  "2": "Sr",
}
`;

exports[`generateDays() Each element matches snapshot 25`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 26`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 27`] = `
Object {
  "0": 3,
  "1": "Czwartek",
  "2": "Cz",
}
`;

exports[`generateDays() Each element matches snapshot 28`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 29`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 30`] = `
Object {
  "0": 4,
  "1": "Piątek",
  "2": "Pt",
}
`;

exports[`generateDays() Each element matches snapshot 31`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 32`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 33`] = `
Object {
  "0": 5,
  "1": "Sobota",
  "2": "Sb",
}
`;

exports[`generateDays() Each element matches snapshot 34`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 35`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 36`] = `
Object {
  "0": 6,
  "1": "Niedziela",
  "2": "Nd",
}
`;

exports[`generateDays() Each element matches snapshot 37`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 38`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 39`] = `
Object {
  "0": 0,
  "1": "Poniedziałek",
  "2": "Pn",
}
`;

exports[`generateDays() Each element matches snapshot 40`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 41`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 42`] = `
Object {
  "0": 1,
  "1": "Wtorek",
  "2": "Wt",
}
`;

exports[`generateDays() Each element matches snapshot 43`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 44`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 45`] = `
Object {
  "0": 2,
  "1": "Sroda",
  "2": "Sr",
}
`;

exports[`generateDays() Each element matches snapshot 46`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 47`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 48`] = `
Object {
  "0": 3,
  "1": "Czwartek",
  "2": "Cz",
}
`;

exports[`generateDays() Each element matches snapshot 49`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 50`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 51`] = `
Object {
  "0": 4,
  "1": "Piątek",
  "2": "Pt",
}
`;

exports[`generateDays() Each element matches snapshot 52`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 53`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 54`] = `
Object {
  "0": 5,
  "1": "Sobota",
  "2": "Sb",
}
`;

exports[`generateDays() Each element matches snapshot 55`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 56`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 57`] = `
Object {
  "0": 6,
  "1": "Niedziela",
  "2": "Nd",
}
`;

exports[`generateDays() Each element matches snapshot 58`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 59`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 60`] = `
Object {
  "0": 0,
  "1": "Poniedziałek",
  "2": "Pn",
}
`;

exports[`generateDays() Each element matches snapshot 61`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 62`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 63`] = `
Object {
  "0": 1,
  "1": "Wtorek",
  "2": "Wt",
}
`;

exports[`generateDays() Each element matches snapshot 64`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 65`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 66`] = `
Object {
  "0": 2,
  "1": "Sroda",
  "2": "Sr",
}
`;

exports[`generateDays() Each element matches snapshot 67`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 68`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 69`] = `
Object {
  "0": 3,
  "1": "Czwartek",
  "2": "Cz",
}
`;

exports[`generateDays() Each element matches snapshot 70`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 71`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 72`] = `
Object {
  "0": 4,
  "1": "Piątek",
  "2": "Pt",
}
`;

exports[`generateDays() Each element matches snapshot 73`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 74`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 75`] = `
Object {
  "0": 5,
  "1": "Sobota",
  "2": "Sb",
}
`;

exports[`generateDays() Each element matches snapshot 76`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 77`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 78`] = `
Object {
  "0": 6,
  "1": "Niedziela",
  "2": "Nd",
}
`;

exports[`generateDays() Each element matches snapshot 79`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 80`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 81`] = `
Object {
  "0": 0,
  "1": "Poniedziałek",
  "2": "Pn",
}
`;

exports[`generateDays() Each element matches snapshot 82`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 83`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 84`] = `
Object {
  "0": 1,
  "1": "Wtorek",
  "2": "Wt",
}
`;

exports[`generateDays() Each element matches snapshot 85`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 86`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 87`] = `
Object {
  "0": 2,
  "1": "Sroda",
  "2": "Sr",
}
`;

exports[`generateDays() Each element matches snapshot 88`] = `
Object {
  "date": Any<Date>,
  "day": Any<Number>,
  "dayOfTheWeek": Any<Array>,
  "month": Any<Array>,
  "year": Any<Number>,
}
`;

exports[`generateDays() Each element matches snapshot 89`] = `
Object {
  "0": 1,
  "1": "Luty",
}
`;

exports[`generateDays() Each element matches snapshot 90`] = `
Object {
  "0": 3,
  "1": "Czwartek",
  "2": "Cz",
}
`;
